src/Tizen.Network.Nsd/Tizen.Network.Nsd/DnssdService.cs

```diff
@@ -253,9 +318,8 @@ public void AddTXTRecord(string key, string value)
             }
 
             byte[] txtValue;
-            ushort txtLength;
-            GetTxtRecord(out txtLength, out txtValue);
-
+            GetTxtRecord(out txtValue);
+            ushort txtLength = (ushort)txtValue.Length;
             ret = Interop.Nsd.Dnssd.SetRecord(_serviceHandle, _dnsRecordtype, txtLength, txtValue);
```

Comment:
I don't understand why `SetRecord` was required after `AddTxtRecord`

Reply:
SetRecord used to set which type of record to add (please see the _dnsRecordtype param in setRecord)